experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,135 @@
+import { DecisionDialogComponent, DecisionDialogData } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * StatusDetailsComponent is the controller for a status "details" page.
+ */
+@Component({
+	selector: "tp-status-details",
+	styleUrls: ["./status-details.component.scss"],
+	templateUrl: "./status-details.component.html",
+})
+export class StatusDetailsComponent {
+	public new = false;
+
+	/** Loader status for the actions */
+	public loading = false;
+
+	/** All details of status requested */
+	public statusDetails!: ResponseStatus;
+
+	/** Reactive form intialized to creat / edit status details */
+	public statusDetailsForm: FormGroup = new FormGroup({
+		description: new FormControl("", Validators.required),
+		name: new FormControl("", Validators.required),
+	});
+
+	/**
+	 * Constructor.
+	 *
+	 * @param api The Servers API which is used to provide row data.
+	 * @param route A reference to the route of this view which is used to get the 'id' query parameter of status.
+	 * @param router Angular router
+	 * @param dialog Dialog manager
+	 * @param fb Form builder
+	 * @param navSvc Manages the header
+	 */
+	constructor(
+		private readonly api: ServerService,
+		private readonly route: ActivatedRoute,
+		private readonly dialog: MatDialog,
+		private readonly navSvc: NavigationService, private readonly location: Location,
+	) {
+		// Getting id from the route
+		const id = this.route.snapshot.paramMap.get("id");
+
+		/**
+		 * Initializes table data, loading it from Traffic Ops.
+		 * we check whether params is a number if not we shall assume user wants to add a new status.
+		 */
+		if (id !== "new") {
+			this.loading = true;
+			this.statusDetailsForm.addControl("id", new FormControl(""));
+			this.statusDetailsForm.addControl("lastUpdated", new FormControl(""));
+			this.getStatusDetails();
+		} else {
+			this.navSvc.headerTitle.next("New Status");
+			this.new = true;
+		}
+	}
+
+	/**
+	 * Get status details for the id
+	 * patch the form with status details
+	 */
+	public async getStatusDetails(): Promise<void> {
+		this.statusDetails = await this.api.getStatuses(this.statusDetails.id);
+
+		// Set page title with status Name
+		this.navSvc.headerTitle.next(`Status #${this.statusDetails.name}`);
+
+		// Patch the form with existing data we got from service requested above.
+		this.statusDetailsForm.patchValue(this.statusDetails);
+		this.loading = false;
+	}
+
+	/**
+	 * On submitting the form we check for whether we are performing Create or Edit
+	 *
+	 * @param event HTML form submission event.
+	 */
+	public async onSubmit(event: Event): Promise<void>  {
+		event.preventDefault();
+		event.stopPropagation();
+
+		if (this.statusDetailsForm.valid) {
+			if (this.new) {
+				this.statusDetails = await this.api.createStatus(this.statusDetailsForm.value);
+				this.location.back();
```

Comment:
Why are we going back to the user's previous location when a new status is created?

Reply:
to show status table page with latest status created as in tpv1 version.

Reply:
That is not done by any other TPv2 details page, so it would be a confusing and inconsistent experience for the user. I don't think it should be done here, either.